.github/workflows/builder_nodejs_slsa3.yml

```diff
@@ -165,13 +179,33 @@ jobs:
           echo "Package path resolved to '${untrusted_package_realpath}'"
           echo "Checking directory '${untrusted_package_realpath}' is a sub-directory of '${github_workspace_realpath}'"
           if [[ "${untrusted_package_realpath}" != "${github_workspace_realpath}" ]] && [[ ${untrusted_package_realpath} != ${github_workspace_realpath}/* ]]; then
-              echo "${untrusted_package_realpath} not a sub-directory of ${GITHUB_WORKSPACE}"
+              >&2 echo "${untrusted_package_realpath} not a sub-directory of ${GITHUB_WORKSPACE}"
               exit 1
           fi
           # Directory was validated. Explicitly trust it.
           package_realpath="${untrusted_package_realpath}"
 
           # Run npm publish using npm fork. We are temporarily using a fork so
           # that we can specify the provenance bundle.
-          # TODO(#1829): Support the --access flag.
-          "$(dirname "$(which node)")"/node_modules/npm/bin/npm publish "${package_realpath}" --access public --provenance "${attestation_realpath}"
+          publish_flags="--provenance ${attestation_realpath}"
+          if [[ "${UNTRUSTED_ACCESS}" != "" ]]; then
+            # We need to validate the input so that it doesn't inject commands or other arguments.
+            # We will let npm validate the input but we will just make sure it's a single alphabetic string.
+            if ! [[ "${UNTRUSTED_ACCESS}" =~ [a-zA-Z]+ ]]; then
```

Comment:
would be nice to have a few unit tests to validate it works properly, e.g., spaces are not accepted

Reply:
For now I added a note to add adversarial tests to #1676